datafusion/physical-plan/src/joins/hash_join.rs

```diff
@@ -747,6 +753,97 @@ where
     Ok(())
 }
 
+// State for storing left/right side indices used for partial batch output
+// & producing ranges for adjusting indices
+#[derive(Debug, Default)]
+pub(crate) struct HashJoinStreamState {
+    // total rows in current probe batch
+    probe_rows: usize,
+    // saved probe-build indices to resume matching from
+    last_matched_indices: Option<(usize, usize)>,
+    // current iteration has been updated
+    matched_indices_updated: bool,
+    // tracking last joined probe side index seen for further indices adjustment
+    last_joined_probe_index: Option<usize>,
+    // tracking last joined probe side index seen for further indices adjustment
```

Comment:
This comment is same as above -- can you clarify what the difference is between "last" and "prev"?

Reply:
Comment has been updated